tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -461,14 +469,48 @@ func (s *storeImpl) Exists(ctx context.Context, {{template "paramList" $pks}}) (
     } else if !ok {
         return false, nil
     }
+    {{- else if .Obj.IsDirectlyScoped }}
+    scopeChecker := sac.GlobalAccessScopeChecker(ctx)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.ResourceWithAccess{
+		Resource: targetResource,
+		Access:   storage.Access_READ_ACCESS,
```

Comment:
In `pkg/auth/permissions/permission.go`, the `View` and `Modify` methods build `ResourceWithAccess` objects from `ResourceMetadata` ones.
In the longer run, it might be good to use these across the codebase rather than manually creating the `ResourceWithAccess` object.

This could be tracked in a follow-up (cleanup) ticket.